geoAssembler/geometry.py

```diff
@@ -268,6 +262,73 @@ def from_h5_file_and_quad_positions(cls, geom_file, quad_pos=None):
         )
         return cls(exgeom_obj, geom_file)
 
+    @classmethod
+    def from_quad_positions(cls, quad_pos):
+        exgeom_obj = cls._extra_geom_from_quad_positions(quad_pos)
+        return cls(exgeom_obj, None)
+
+    @staticmethod
+    def _extra_geom_from_quad_positions(quad_pos, *, unit=1e-3):
```

Comment:
This method is a copy of the code from https://github.com/European-XFEL/EXtra-geom/pull/89